File: VB/Grid_BottomNewItemRow/GridSplitContainerDescendant/GridSplitContainerDescendant.vb
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraGrid
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace Grid_BottomNewItemRow
	Public Class GridSplitContainerDescendant
		Inherits GridSplitContainer

		Public Sub New()
			MyBase.New()
		End Sub

		Protected Overrides Function CreateContainerInfo() As SplitContainerViewInfo
			Return New SplitContainerViewInfoDescendant(Me)
		End Function

		Protected Overrides Function CreateGridControl() As GridControl
			Return New GridControlDescendant()

		End Function

	End Class
End Namespace

File: CS/Grid_BottomNewItemRow/GridSplitContainerDescendant/GridSplitContainerDescendant.cs
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraGrid;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Grid_BottomNewItemRow
{
    public class GridSplitContainerDescendant : GridSplitContainer
    {
        public GridSplitContainerDescendant()
            : base()
        { }

        protected override SplitContainerViewInfo CreateContainerInfo()
        {
            return new SplitContainerViewInfoDescendant(this);
        }

        protected override GridControl CreateGridControl()
        {
            return new GridControlDescendant();
           
        }

    }
}
